process/transaction/shardProcess.go

```diff
@@ -730,6 +719,7 @@ func (txProc *txProcessor) getUserTxCost(
 				"gas used", gasUsed,
 			)
 
+			// TODO: always use all the gas?
```

Comment:
this is changed in gasPrice-modifier - all gas should be consumed in case of any error.